### PR TITLE
allowed special characters in address field

### LINKED ISSUE #1037: Edit Address Text field to also take special characters like (‘,’ ,’.’, ’/’, ’-’)
## Description
In the address field, the text is not allowed to enter the symbolic values like `,`, `.`, `/`, `-`. Add appropriate changes to field along with consistency in database to allow user to insert address having these symbols.

### Update [Required]
- [ ] User should be able to input address in any desired form
- [ ] Consistency with database should me maintained

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor.

## Estimation
2 hours

### PR DESCRIPTION
**Pull request by vaibhavarora102**
# Description
allowed the special characters in the address field, using RegexValidator

Fixes #1037

# Type of Change:

- Code
- Quality Assurance

**Code/Quality Assurance Only**
- Bug fix (non-breaking change which fixes an issue)
- This change requires a documentation update (software upgrade on readme file)
- New feature (non-breaking change which adds functionality pre-approved by mentors)

# How Has This Been Tested?
I had tested on local setup and it works fine

# Checklist:
**Delete irrelevant options.**

- [ ] I have performed a self-review of my own code or materials
- [ ] Any dependent changes have been merged 

**Code/Quality Assurance Only**
- [ ] My changes generate no new warnings 
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by keshakaneria** [comment]
You might have removed the end new line in the file because of which this build is failing. Try adding a new at the end of file and push the commit again @vaibhavarora102

**Comment by vaibhavarora102** [comment]
I did not understand can you @keshakaneria please explain a bit as i think i had not removed anything

**Comment by keshakaneria** [comment]
> I did not understand can you @keshakaneria please explain a bit as i think i had not removed anything

Just once in your editor if there is any newline at the end of this file. If No, then add a new blank line at the end of the file in which you have made changes.

**Comment by vaibhavarora102** [comment]
no, I checked there is an extra line at the end

**Comment by keshakaneria** [comment]
@Kajol-Kumari Can you let us know why still this build is failed?

**Comment by Kajol-Kumari** [comment]
Hey @keshakaneria @vaibhavarora102 The build is failing due to [this](https://github.com/anitab-org/vms/blob/develop/vms/administrator/tests/test_unit.py#L192) test. You can get to know more about the error from [here](https://travis-ci.com/github/anitab-org/vms/builds/184069541#L905) to [here](https://travis-ci.com/github/anitab-org/vms/builds/184069541#L910) . As you have modified the address validation regex so you need to modify this test accordingly.

**Comment by keshakaneria** [comment]
> Hey @keshakaneria @vaibhavarora102 The build is failing due to [this](https://github.com/anitab-org/vms/blob/develop/vms/administrator/tests/test_unit.py#L192) test. You can get to know more about the error from [here](https://travis-ci.com/github/anitab-org/vms/builds/184069541#L905) to [here](https://travis-ci.com/github/anitab-org/vms/builds/184069541#L910) . As you have modified the address validation regex so you need to modify this test accordingly.

Thanks for this :)

**Comment by vaibhavarora102** [comment]
I think they are saying to correct the test and I guess this could be done by you 
Correct me if I am wrong

**Comment by keshakaneria** [comment]
> I think they are saying to correct the test and I guess this could be done by you
> Correct me if I am wrong

The changes should be tested by you before pushing. Thus, when you will try to test the changes, this error will show up. Let me know if this error shows up in your local system.

**Comment by vaibhavarora102** [comment]
on local host it is working good
kindly tell what next to do

**Comment by vaibhavarora102** [comment]
Hi any updates?

**Comment by keshakaneria** [comment]
> Hi any updates?

Ahh... I'm sorry, I just skipped this issue to review. Can you send the the screenshot which shows that all the tests are passed when you try to run the tests locally?

File: vms/administrator/models.py [hunk 1]
```diff
@@ -21,7 +21,7 @@ class Administrator(models.Model):
     address = models.CharField(
         max_length=75,
         validators=[
-            RegexValidator(r'^[(A-Z)|(a-z)|(0-9)|(\s)|(\-)|(\.)|(,)|(\:)]+$', ),
+            RegexValidator(r'^[(A-Z)|(a-z)|(0-9)|(\s)|(\-)|(\.)|(,)|(\:)|(@_!#$%^&*()<>?/\|}{~:)]+$', ),
```
Comment (SanketDG) [inline vms/administrator/models.py:24]: This is too extensive, I think @keshakaneria's exact list on the issue description is good enough, please follow that, I don't think I have seen `$`, `%` in any address.

**Review by SanketDG — CHANGES_REQUESTED** [review]

File: vms/administrator/models.py (hunk 1 above)
Comment (vaibhavarora102) [reply vms/administrator/models.py]: done
kindly check

File: vms/administrator/models.py [hunk 2]
```diff
@@ -21,7 +21,7 @@ class Administrator(models.Model):
     address = models.CharField(
         max_length=75,
         validators=[
-            RegexValidator(r'^[(A-Z)|(a-z)|(0-9)|(\s)|(\-)|(\.)|(,)|(\:)|(@_!#$%^&*()<>?/\|}{~:)]+$', ),
+            RegexValidator(r'^[(A-Z)|(a-z)|(0-9)|(\s)|(\-)|(\.)|(,)|(\:)|(/\.,-_)]+$', ),
```
Comment (SanketDG) [inline vms/administrator/models.py:24]: Here's a much simple shorter one 
```suggestion
            RegexValidator(r'^[A-Za-z0-9\s\-.,_/:]+$', ),
```

**Comment by keshakaneria** [comment]
Include these suggested changes @vaibhavarora102

**Comment by vaibhavarora102** [comment]
I missed that comment
Thanks for reminding
I will surely do it by today

**Comment by vaibhavarora102** [comment]
kindly check @keshakaneria

**Review by keshakaneria — APPROVED** [review]
Now it looks good 💯

**Review by SanketDG — APPROVED** [review]
LGTM!

**Comment by vaibhavarora102** [comment]
Hi @keshakaneria  when will my pr got merged as I need to fill OSH form

**Comment by keshakaneria** [comment]
> Hi @keshakaneria when will my pr got merged as I need to fill OSH form

Yes @vaibhavarora102. The number of PR opened which gets finally merged before or after evaluation will be counted. Since, this PR has `ready-to-merge` tag, it will be merged soon by maintainers. You can proceed to fill the form.

**Comment by vaibhavarora102** [comment]
Thanks

**Review by Kajol-Kumari — APPROVED** [review]
Good to go now! 
Please squash the commits so that it can be merged.

**Comment by vaibhavarora102** [comment]
Hi, @keshakaneria  while squashing i am facing some problem can i create another pr and close this one in respect to this

**Comment by keshakaneria** [comment]
> Hi, @keshakaneria while squashing i am facing some problem can i create another pr and close this one in respect to this

This is even failing the test [jobDetails](https://travis-ci.com/github/anitab-org/vms/builds/188101594#L913) and [EventDetails](https://travis-ci.com/github/anitab-org/vms/builds/188101594#L892)

**Pull request closed without merging**